src/XMakeBuildEngine/UnitTests/BuildEnvironmentHelper_Tests.cs

```diff
@@ -105,7 +105,7 @@ public void BuildEnvironmentDetectsVisualStudioByEnvironment()
             {
                 Environment.SetEnvironmentVariable("VSINSTALLDIR", env.TempFolderRoot);
                 Environment.SetEnvironmentVariable("VisualStudioVersion", "15.0");
-                BuildEnvironmentHelper.ResetInstance_ForUnitTestsOnly();
+                BuildEnvironmentHelper.ResetInstance_ForUnitTestsOnly(getAppContextBaseDirectory: ReturnNull);
```

Comment:
I was worried if the order ever changed then the test could start to fail so I put this in here to remove the AppContext.BaseDirectory from the equation for all cases.  I can remove it if you'd like